src/components/networked-video-source.js

```diff
@@ -0,0 +1,87 @@
+/* global AFRAME, NAF, THREE */
+var naf = require('../NafIndex');
+
+AFRAME.registerComponent('networked-video-source', {
+
+  schema: {
+  },
+
+  dependencies: ['material'],
+
+  init: function () {
+    this.videoTexture = null;
+    this.video = null;
+    this.stream = null;
+
+    this._setMediaStream = this._setMediaStream.bind(this);
+
+    NAF.utils.getNetworkedEntity(this.el).then((networkedEl) => {
+      const ownerId = networkedEl.components.networked.data.owner;
+
+      if (ownerId) {
+        NAF.connection.adapter.getMediaStream(ownerId, "video")
+          .then(this._setMediaStream)
+          .catch((e) => naf.log.error(`Error getting media stream for ${ownerId}`, e));
+      } else {
+        // Correctly configured local entity, perhaps do something here for enabling debug audio loopback
+      }
+    });
+  },
+
+  _setMediaStream(newStream) {
+
+    if(!this.video) {
+      this.setupVideo();
+    }
+
+    if(newStream != this.stream) {
+      if (this.stream) {
+        this._clearMediaStream();
+      }
+
+      if (newStream) {
+        this.video.srcObject = newStream;
```

Comment:
Maybe create a new stream with explicitly only the first video track before setting srcObject?
Not tested:
```
newStream = new MediaStream(newStream.getVideoTracks()[0]);
this.video.srcObject = newStream;
```
maybe make sure we have a least a video track.

Reply:
I already use "NAF.connection.adapter.getMediaStream(ownerId, "video")" with video parameter for that and call _setMediaStream. if null it will clear previous if there was any, otherwise do nothing and set stream to null.